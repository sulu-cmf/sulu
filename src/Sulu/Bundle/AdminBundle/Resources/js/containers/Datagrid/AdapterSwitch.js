// @flow
import React from 'react';
import ButtonGroup from '../../components/ButtonGroup';
import Button from '../../components/Button';
import Icon from '../../components/Icon';
import datagridAdapterRegistry from './registries/DatagridAdapterRegistry';

type Props = {
    adapters: Array<string>,
    currentAdapter: string,
    onAdapterChange: (adapter: string) => void,
};

export default class AdapterSwitch extends React.PureComponent<Props> {
    handleAdapterChange = (adapter: ?string) => {
        if (!adapter || this.props.currentAdapter === adapter) {
            return;
        }

        this.props.onAdapterChange(adapter);
    };

    render() {
        const {
            currentAdapter,
            adapters,
        } = this.props;

        if (adapters.length < 2) {
            return null;
        }

        return (
            <ButtonGroup>
                {adapters.map((adapter, index) => {
                    const Adapter = datagridAdapterRegistry.get(adapter);

                    return (
                        <Button
                            active={adapter === currentAdapter}
                            key={index}
                            onClick={this.handleAdapterChange}
                            value={adapter}
                        >
                            <Icon name={Adapter.icon} />
                        </Button>
                    );
                })}
            </ButtonGroup>
        );
    }
}
