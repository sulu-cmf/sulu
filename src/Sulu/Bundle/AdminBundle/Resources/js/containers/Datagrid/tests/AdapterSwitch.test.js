// @flow
import {mount} from 'enzyme';
import React from 'react';
import AdapterSwitch from '../AdapterSwitch';

test('The component should render with current adapter "folder"', () => {
    const adapters = ['table', 'folder'];
    const currentAdapterKey = 'folder';
    const handleAdapterChange = jest.fn();
    const view = mount(
        <AdapterSwitch
            adapters={adapters}
            currentAdapter={currentAdapterKey}
            onAdapterChange={handleAdapterChange}
        />
    ).render();

    expect(view).toMatchSnapshot();
});

test('The component should render with current adapter "table"', () => {
    const adapters = ['table', 'folder'];
    const currentAdapterKey = 'table';
    const handleAdapterChange = jest.fn();
    const view = mount(
        <AdapterSwitch
            adapters={adapters}
            currentAdapter={currentAdapterKey}
            onAdapterChange={handleAdapterChange}
        />
    ).render();

    expect(view).toMatchSnapshot();
});
