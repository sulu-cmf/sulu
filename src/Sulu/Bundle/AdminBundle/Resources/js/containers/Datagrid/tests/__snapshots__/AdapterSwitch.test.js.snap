// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The component should render with current adapter "folder" 1`] = `
<ul>
  <li>
    table
  </li>
  <li>
    <i>
      folder
    </i>
  </li>
</ul>
`;

exports[`The component should render with current adapter "table" 1`] = `
<ul>
  <li>
    <i>
      table
    </i>
  </li>
  <li>
    folder
  </li>
</ul>
`;
